wikidict/render.py

```diff
@@ -0,0 +1,386 @@
+    for section in parsed.get_sections(include_subsections=True, level=level):
+        title = section.title.replace(" ", "").lower()
+        if title not in head_sections[locale]:
+            continue
+
+        for sublevel in section_sublevels[locale]:
+            sections.extend(
+                section.get_sections(include_subsections=False, level=sublevel)
+            )
+
+    return sections
+
+
+def find_sections(code: str, locale: str) -> Sections:
+    """Find the correct section(s) holding the current locale definition(s)."""
+    ret = defaultdict(list)
+    for section in find_all_sections(code, locale):
+        title = section.title.strip()
+        if not title.startswith(sections[locale]):
+            continue
+        ret[title].append(section)
+    return ret
+
+
+def find_titles(code: str, locale: str) -> List[str]:
+    """Find the correct section(s) holding the current locale definition(s)."""
+    return [
+        section.title.strip()
+        for section in find_all_sections(code, locale)
+        if section.title
+    ]
+
+
+def parse_word(word: str, code: str, locale: str, force: bool = False) -> Word:
+    """Parse *code* Wikicode to find word details.
+    *force* can be set to True to force the pronunciation and genre guessing.
+    It is disabled by default t spee-up the overall process, but enabled when
+    called from get_and_parse_word().
+    """
+    parsed_sections = find_sections(code, locale)
+    prons = []
+    nature = ""
+    etymology = ""
+
+    # Etymology
+    sections = etyl_section[locale]
+    if not isinstance(sections, list):
+        sections = [sections]  # type: ignore
+    for section in sections:
+        etyl_data = parsed_sections.pop(section, [])
+        if etyl_data:
+            etymology = find_etymology(word, locale, etyl_data[0])
+
+    definitions = find_definitions(word, parsed_sections, locale)
+
+    if definitions or force:
+        prons = find_pronunciations(code, pronunciation[locale])
+        nature = find_genre(code, genre[locale])
+
+    # find if variant and delete unwanted definitions
+    variants = []
+    if locale == "fr" and not definitions:  # Pure verb form, no definition
+        for title, s in parsed_sections.items():
+            if title.startswith("{{S|verbe|fr|flexion"):
+                for t in s[0].templates:
+                    if t.__str__().startswith("{{fr-verbe-flexion"):
+                        infinitive = process_templates(word, clean(t.__str__()), locale)
+                        variants.append(infinitive)
+
+    return Word(prons, nature, etymology, definitions, variants)
+
+
+def get_latest_json_file(locale: str, output_dir: Path) -> str:
+    """Get the name of the last data-*.json file."""
+    files = os.listdir(output_dir)
+    files = [f for f in files if f.startswith("data_wikicode-") and f.endswith(".json")]
+    if not files:
+        return ""
+    files = sorted(files)
+    return files[-1]
```

Comment:
```suggestion
    files = list(output_dir.glob("data_wikicode-*.json")
    return sorted(files)[-1] if files else ""
```

Reply:
Do you plan to apply those changes?

Reply:
Sure, checking pool.map now but I will do it next.

Reply:
Oh you are quick :)
That could be done in another PR if you prefer. LMK when you are ready for a final review.

Reply:
Not so quick :) I couldn't make it work with pool.map yet :(

Reply:
Done and squashed.